### PR TITLE
Adds information on CLI on Docker

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR adds information for how to run Zonemaster-CLI on Docker.

* Adds information on Docker in the USING document
* Corrects the information in the USING document
* Adds table of contents in the USING document
* Adds information of translation in the USING document.
* Removes peripheral information from the USING document
* Removes detailed Docker build documentation and refers to other document.
* Adds referral to the USING document for Docker use.
* Adds table of contents to the Installation document.
* Adds referral to USING for Docker use in the Installation document.

## Changes

The following documents are updated:

* README.md
* USING.md
* docs/Installation.md

## How to test this PR

In this PR only documentation is updated. By following the instructions it could be verified.

Note that this PR adds a link to a document in the master branch that is now only available on the develop branch, but will also be available on the master branch at release.
```
https://github.com/zonemaster/zonemaster/blob/master/docs/internal-documentation/maintenance/ReleaseProcess-create-docker-image.md
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,84 +1,57 @@
...
+## Docker
 
-```sh
-make docker-tag-latest
-```
+Zonemaster-CLI is available on [Docker Hub], and can be run without any
+installation. See [USING] Zonemaster-CLI for how to run it on Docker.
 
-Test a zone using the local base image:
+To build your own Docker image, see [Create Docker Image] documentation.
```
Comment (mattias-p) [inline README.md:35]: Could this piece of documentation have a noun phrase for its name instead of a verb phrase? E.g. "Docker Image Creation".

File: USING.md [hunk 2]
```diff
@@ -1,116 +1,210 @@
...
-like this:
 
-    $ zonemaster-cli example.com
+## Docker or local installation
 
-The output comes continously as the tests are performed.
+The `zonemaster-cli` tool can be run from the command line of any computer that
+meets one of the following requirements:
+
+* Docker is installed on the computer, or
```
Comment (mattias-p) [inline USING.md:16]: You need privileges too for Docker.

File: USING.md [hunk 3]
```diff
@@ -1,116 +1,210 @@
...
-      13.86 CRITICAL  All nameservers are in the same AS (12345).
-      13.87 NOTICE    123.456.789.0 returned no DS records for example.com.
+### Using Docker
 
-If your machine is for some reason not configured for use with IPv6 you
-want to disable the use of IPv6 with the `--no-ipv6` option. If you want
-to see how your domain is configured to be used with no IPv4 available
-for the client there is also the `--no-ipv4` option - however, both cannot
-be used at the same time.
+To run Zonemaster-CLI on docker you have to make sure that Docker is installed.
```
Comment (mattias-p) [inline USING.md:21]: Is there a reason why this one reference to Docker isn't capitalized when all the others are?

File: USING.md [hunk 4]
```diff
@@ -1,116 +1,210 @@
...
+```
 
-More detailed documentation can by found in the [manual page](zonemaster-cli.md)
+With local installation:
+```sh
+zonemaster-cli zonemaster.net
+```
 
-### Advanced use
+The output comes continously as the tests are performed.
```
Comment (mattias-p) [inline USING.md:47]: `continously` -> `continuously`

File: USING.md [hunk 5]
```diff
@@ -1,116 +1,210 @@
...
+
+```sh
+docker run -t --rm zonemaster/cli zonemaster.net --no-ipv6 --show-testcase --locale=da
+```
+```sh
+zonemaster-cli zonemaster.net --no-ipv6 --show-testcase --locale=da_DK.UTF-8
+```
+
+The difference between running `zonemaster-cli` on Docker or local installation
+is invocation string, `docker run -t --rm zonemaster/cli` vs. `zonemaster-cli`.
```
Comment (mattias-p) [inline USING.md:76]: `is invocation string` -> `is the invocation string`

File: USING.md [hunk 6]
```diff
@@ -1,116 +1,210 @@
...
+To simplify this document, from now on the short `zonemaster-cli` will be used
```
Comment (mattias-p) [inline USING.md:77]: `short` -> `shorter` maybe?

File: USING.md [hunk 7]
```diff
@@ -1,116 +1,210 @@
...
+
+The severity level of the different messages is CRITICAL, ERROR, WARNING, NOTICE,
+INFO, DEBUG, DEBUG2 or DEBUG3. The default reporting level is NOTICE and higher.
+To change the level of reporting you can use a command line option, e.g
+`--level=INFO` includes level INFO and higher in the report. See
+"[Severity Level Definitions]" for more information on the levels.
+
+The default level reporting is in plain English (or some other language), but
+other output formats are also available with options `--raw` and `json`
+respectively.
```
Comment (mattias-p) [inline USING.md:97]: `The default level reporting is in plain English (or some other language)` -> `By default the output is formatted as plain text` maybe?

File: USING.md [hunk 8]
```diff
@@ -1,116 +1,210 @@
...
+es-CL.UTF-8 | Spanish
+sv_SE.UTF-8 | Swedish
+
+E.g.:
+```sh
+docker run -t --rm zonemaster/cli zonemaster.net --locale=da_DK.UTF-8
+```
+
+If the environment variable `LANGUAGE` is set with correct LOCALE then no option
+is needed, e.g. `LANGUAGE=da_DK.UTF-8`.
```
Comment (mattias-p) [inline USING.md:153]: Shouldn't we be saying that zonemaster-cli respects all the environment variables LANG, LC_MESSAGES, LC_ALL and LANGUAGE?

File: USING.md [hunk 9]
```diff
@@ -1,116 +1,210 @@
...
+```sh
+zonemaster-cli --list_tests
+```
+
+## Undelegated test
+
+Before you do any delegation change at the parent, either changing the NS
+records, glue address records or DS records, you might want to perform a
+check of your new child zone configuration so that everything you plan to
+change is in order. Zonemaster can do this for your. All you have to do
```
Comment (mattias-p) [inline USING.md:187]: `for your` -> `for you`

**Review by mattias-p — COMMENTED** [review]
It's great to see the USING document be updated. I think its new contents looks really good too.

File: USING.md [hunk 10]
```diff
@@ -1,116 +1,216 @@
...
+E.g.:
+```sh
+docker run -t --rm zonemaster/cli zonemaster.net --locale=da
+```
+
+An alternative is to set the `LC_ALL` environment variable with correct language
+value when the command is invoked, which can be useful if a shell alias is
+created. E.g.
+```sh
+docker run -e LC_ALL=da -t --rm zonemaster/cli zonemaster.net
```
Comment (blacksponge) [inline USING.md:131]: Also passing the LANG / LC_ALL environment without value will take the value from the one defined at run time. 
So `docker run -e LANG -t --rm zonemaster/cli zonemaster.net` will use the default locale defined on the user environment. I don't know if it worth mentioning.

File: USING.md [hunk 11]
```diff
@@ -1,116 +1,216 @@
...
-your, all you have to do is give Zonemaster all the parent data you
-plan to have for your new configuration. Any DNS lookups going for
-the parent will instead be answered by the data you entered.
+If you only want to run a specific test case rather than the whole suite of
+tests, you can do that as well. E.g. test only test case [Connectivity03]:
 
-    $ zonemaster-cli --ns ns1.example.com/192.168.23.23 \
-          --ns ns2.example.com/192.168.24.24 \
-          --ds 12345,3,1,123456789abcdef67890123456789abcdef67890
+[Connectivity03]:  https://github.com/zonemaster/zonemaster/blob/master/docs/specifications/tests/Connectivity-TP/connectivity03.md
```
Comment (blacksponge) [inline USING.md:171]: Why not put this link at the end of the file?

**Review by blacksponge — COMMENTED** [review]
nice documentation improvement!

File: USING.md (hunk 10 above)
Comment (matsduf) [reply USING.md:131]: I have updated with that information, but maybe too many details?

File: USING.md (hunk 11 above)
Comment (matsduf) [reply USING.md]: My mistake. Thanks for catching it.

File: USING.md (hunk 10 above)
Comment (blacksponge) [reply USING.md:131]: I think it looks fine

File: USING.md (hunk 5 above)
Comment (matsduf) [reply USING.md]: Fixed.

**Comment by matsduf** [comment]
@blacksponge, please approve if you are fine with this PR.

**Review by blacksponge — APPROVED** [review]

**Review by mattias-p — APPROVED** [review]
